src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -124,18 +124,18 @@ protected override async Task<SearchResult> SearchInternalAsync(SearchOptions se
             }
             else
             {
-                searchResult = await SearchImpl(searchOptions, false, cancellationToken);
+                searchResult = await SearchImpl(searchOptions, false, false, null, cancellationToken);
             }
 
             return searchResult;
         }
 
         protected override async Task<SearchResult> SearchHistoryInternalAsync(SearchOptions searchOptions, CancellationToken cancellationToken)
         {
-            return await SearchImpl(searchOptions, true, cancellationToken);
+            return await SearchImpl(searchOptions, true, false, null, cancellationToken);
         }
 
-        private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool historySearch, CancellationToken cancellationToken)
```

Comment:
>bool historySearch [](start = 81, length = 18)

Maybe it's worth to introduce enum: normal, history, reindex instead of adding new bool value. At least if the mutually exclusive.

Reply:
Ah yes, that would look a bit cleaner - but you're right that it would give the impression that `isHistorySearch` and `isReindexSearch` are mutually exclusive, which they are right now but will not always be.

A bit more info: we have (far off) plans to reindex resource history once we enable fine-grained access control policies on historical data. To support that, we will need to have both of these flags set to true for reindex searches.

Reply:
Well, even if they not mutually excusive, you still can go with enum, just make sure it's marked with `[Flags]` attribute [https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-5.0](https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-5.0)

But I don't insist :)

---
In reply to: [577119723](https://github.com/microsoft/fhir-server/pull/1668#discussion_r577119723) [](ancestors = 577119723)

Reply:
Nice didn't know we could add that attribute to enums. Let me know how ef75010 looks.